### DISCUSSION TITLE
@media css

### DISCUSSION THREAD

**Discussion by PaulMcF1987** [Budibase/budibase#7276] [Help]
How do I create different layouts for mobile/desktop?

The buttons at the top allow you to view how the page will look on different devices, but I cannot see any way to change the CSS depending on the device.

Usually, I would use `@media...` but in order to do this I need to know the Class/ID associated with each element.

**Comment by R2bEEaton** [comment]
Budibase is already fairly good at changing styles based on device, but if something is becoming a problem, you can use [component Conditions](https://docs.budibase.com/docs/conditions).

Another important thing to note is that there are global bindings `{{ Device.Mobile }}` and `{{ Device.Tablet }}` that you can use in these Conditions or anywhere else.

For example, you could set a condition to make a cards block go vertically instead of horizontally when `{{ Device.Mobile }}` is `true`.

If this isn't good enough, you could even make two containers with totally different layouts. Use Conditions to completely hide one container when `{{ Device.Mobile }}` is `true` and show it when it's false, and vice versa with the other container. Now you have totally separate wide screen and mobile layouts. This might be annoying to do because you would need to updated two layouts whenever you need to make changes, so should be avoided.

Finally, if none of this is good enough, you could use `@media...` as you mentioned by either changing the style of a component or using an embed element with a `<style>` tag. To reference an element, you can set the "Name" field in Budibase. This sets the HTML element's `data-name` attribute, and can therefore be used in CSS with `[data-name="table1"] { ... }`. Make sure to have Inspect Element handy.

Hopefully this helps!

**Reply by PaulMcF1987** [reply]
@R2bEEaton the thing I am more looking for is... 

I might have it set on a screen for a container to be horizontal and the width set as 50vw for the 2 components inside the container...

Then on a mobile device Id want the container to be vertical and have the width of the elements inside to be 100vw instead of 50vw.

I cant seem to find a way to achieve the width part of the problem

**Reply by R2bEEaton** [reply]
I see. Yeah, the first part is do-able with conditions, but unfortunately not the width part as far as I know. What are the two components inside? Just having a hard time understanding what you're trying to do specifically. If it's a paragraph or something similar, you could get away with something like this. It switches to a vertical mode with completely default settings.

The section is set to two columns:
![image](https://user-images.githubusercontent.com/34921506/184668379-aaa8ed82-052b-4e91-bd6c-c0c3323fe863.png)

![image](https://user-images.githubusercontent.com/34921506/184668444-57d07fa7-81e1-437a-af88-65dc9b0dca22.png)
![image](https://user-images.githubusercontent.com/34921506/184668465-8d75b071-3fd8-4919-8833-512119effabc.png)

**Reply by PaulMcF1987** [reply]
@R2bEEaton it can vary depending on the app or the page of the app.

It is something that I have come up against on quite a few occasions... the app looks decent on a desktop but looks messy on a mobile device... I then tidy it up on the mobile and it looks messy on the desktop.

There is only one option to set size (Width and height)... If I was creating something myself, I would have a stylesheet with different @media styles set so that it automatically resized based on the device. Or I could use Bootstrap CSS

Im basically just looking for a way to set the size (height/width) based on the device

**Reply by R2bEEaton** [reply]
Right, I guess unfortunately the only option for you right now would be an embed component with custom styles like I mentioned above, sorry.
